test/e2e/gardener/shoot/create_authenticate_delete.go

```diff
@@ -32,179 +32,193 @@ import (
-
-		shoot2TokenClientAPIServerProxy, err := getAPIServerProxyClient(shoot2TokenClient.RESTConfig(), shoot2)
-		Expect(err).NotTo(HaveOccurred())
-
-		Eventually(func(g Gomega) {
-			g.Expect(validateShootAccess(ctx, shoot1TokenClientAPIServerProxy, shoot1, true)).To(Succeed())
-			g.Expect(validateShootAccess(ctx, shoot2TokenClientAPIServerProxy, shoot2, true)).To(Succeed())
-		}).WithTimeout(time.Minute).Should(Succeed())
-
-		By("Verify a shoot cannot be accessed with a service account token from another shoot by manipulating the apiserver-proxy header")
-		shoot1NoAccessTokenClientAPIServerProxy, err := getAPIServerProxyClient(shoot1NoAccessTokenRestConfig, shoot1)
-		Expect(err).NotTo(HaveOccurred())
-
-		shoot2NoAccessTokenClientAPIServerProxy, err := getAPIServerProxyClient(shoot2NoAccessTokenRestConfig, shoot2)
-		Expect(err).NotTo(HaveOccurred())
-
-		Consistently(func(g Gomega) {
-			g.Expect(validateShootAccess(ctx, shoot1NoAccessTokenClientAPIServerProxy, shoot1, false)).To(Succeed())
-			g.Expect(validateShootAccess(ctx, shoot2NoAccessTokenClientAPIServerProxy, shoot2, false)).To(Succeed())
-		}).WithTimeout(10 * time.Second).Should(Succeed())
-
-		By("Verify that authentication with istio tls termination cannot be bypassed")
-		externalAddress, httpClient, err := httpClientForRESTConfig(shoot1Client.RESTConfig())
-		Expect(err).To(Succeed())
-
-		httpRequest, err := http.NewRequestWithContext(ctx, http.MethodGet, externalAddress, http.NoBody)
-		Expect(err).To(Succeed())
-		httpRequest.Header = map[string][]string{
-			"X-Remote-User":  {"kubernetes-admin"},
-			"X-Remote-Group": {"system:masters"},
-		}
+		var shoot1Client, shoot2Client, shoot1TokenClient, shoot2TokenClient kubernetes.Interface
 
-		var httpResponse *http.Response
-		Eventually(func(g Gomega) {
+		It("should verify shoot access using admin kubeconfig", func(ctx SpecContext) {
 			var err error
-			httpResponse, err = httpClient.Do(httpRequest)
-			g.Expect(err).To(Succeed())
-		}).Should(Succeed())
 
-		Expect(httpResponse.StatusCode).To(Equal(http.StatusUnauthorized))
+			shoot1Client, err = access.CreateShootClientFromAdminKubeconfig(ctx, shoot1.GardenClientSet, shoot1.Shoot)
+			Expect(err).NotTo(HaveOccurred())
 
-		By("Verify that users cannot escalate their privileges with istio tls termination")
-		customClient, err := viewerClientWithCustomTransport(ctx, f.GardenClient, shoot1, &injectHeaderTransport{
-			Headers: map[string]string{
-				"X-Remote-User":  "fake-kubernetes-admin",
-				"X-Remote-Group": "system:masters",
-			},
+			shoot2Client, err = access.CreateShootClientFromAdminKubeconfig(ctx, shoot2.GardenClientSet, shoot2.Shoot)
+			Expect(err).NotTo(HaveOccurred())
+
+			Eventually(func(g Gomega) {
+				g.Expect(validateShootAccess(ctx, shoot1Client.Client(), shoot1.Shoot, true)).To(Succeed())
+				g.Expect(validateShootAccess(ctx, shoot2Client.Client(), shoot2.Shoot, true)).To(Succeed())
+			}).WithTimeout(time.Minute).Should(Succeed())
```

Comment:
Any particular reason why this timeout is so much bigger than the ones below?

Reply:
In #11085 I increased the wrong timeout to wait for the token client to be ready. This issue should be solved now, since we are using the dynamic token client now.
I just forgot to decrease the timeout again 😅